core/src/impl/Kokkos_Utilities.hpp

```diff
@@ -409,13 +410,68 @@ struct inclusive_scan_integer_sequence {
   using value_type = typename IntegerSequence::value_type;
   using helper     = inclusive_scan_integer_sequence_helper<
       reverse_integer_sequence<IntegerSequence>,
-      std::integral_constant<value_type, 0> >;
+      std::integral_constant<value_type, 0>>;
   using type                        = typename helper::type;
   static constexpr value_type value = helper::value;
 };
 
 template <typename T>
-using identity_t = T;
+struct identity {
+  using type = T;
+};
+
+template <typename T>
+using identity_t = typename identity<T>::type;
+
+//==============================================================================
+// <editor-fold desc="Folding emulation"> {{{1
+
+// acts like void for comma fold emulation
+struct _fold_comma_emulation_return {};
+
+template <class... Ts>
+constexpr KOKKOS_INLINE_FUNCTION _fold_comma_emulation_return
+emulate_fold_comma_operator(Ts&&...) noexcept {
+  return _fold_comma_emulation_return{};
+}
+
+#define KOKKOS_IMPL_FOLD_COMMA_OPERATOR(expr)                                \
+  ::Kokkos::Impl::emulate_fold_comma_operator(                               \
+      ::std::initializer_list<::Kokkos::Impl::_fold_comma_emulation_return>{ \
+          ((expr), ::Kokkos::Impl::_fold_comma_emulation_return{})...})
+
+// </editor-fold> end Folding emulation }}}1
+//==============================================================================
+
+//==============================================================================
+// <editor-fold desc="remove_cvref_t"> {{{1
```

Comment:
I do not want to bike shed this, but I am voting that we don't put things for specific editors / plugins unless it is widely agreed that the benefits outweigh the noise.  Not really looking to discuss the merits of these comments, just saying that I vote for removal.

Reply:
Just for the record here, since the follow up to this discussion was on Slack: this discussion has been had before. This comment format works with every editor I've used (it's a combination of vim fold markers and eclipse fold tags; pretty much every editor supports one or the other, if not both). I won't rehash the whole discussion here, but code folding is a good way to organize complicated code and a lot of people use it to better comprehend large chunks of code.